File: 250222/최대공약수 구하기/find-the-greatest-common-divisor.js
const fs = require("fs");

const [a, b] = fs.readFileSync(0, "utf8").trim().split(/\s+/).map(Number);

function gcd(a, b) {
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

console.log(gcd(a, b));

File: 250222/최대공약수 구하기/find-the-greatest-common-divisor.cpp
#include <iostream>

using namespace std;

int gcd(int n, int m)
{
    if (n % m == 0)
        return m;
    return gcd(m, n % m);
}

int main() {
    int n, m;
    cin >> n >> m;

    cout << gcd(n, m);
    return 0;
}